Description: Add jcommander to the classpath when TestNG is used.
 This fixes a ClassNotFoundException when running the OpenJDK 8 tests
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: no
--- a/src/share/classes/com/sun/javatest/regtest/exec/MainAction.java
+++ b/src/share/classes/com/sun/javatest/regtest/exec/MainAction.java
@@ -570,6 +570,7 @@
                 .append(jdk.getJDKClassPath())
                 .append(script.getJUnitPath())
                 .append(script.getTestNGPath())
+                .append(new SearchPath("/usr/share/java/jcommander.jar"))
                 .asList();
 
         Version v = script.getRequiredVersion();
--- a/src/share/classes/com/sun/javatest/regtest/exec/RegressionScript.java
+++ b/src/share/classes/com/sun/javatest/regtest/exec/RegressionScript.java
@@ -811,16 +811,20 @@
                 File md = workDir.getFile("modules");
                 if (needJUnit)
                     install(params.getJUnitPath(), md);
-                if (needTestNG)
+                if (needTestNG) {
                     install(params.getTestNGPath(), md);
+                    install(new SearchPath("/usr/share/java/jcommander.jar"), md);
+                }
                 mp.append(md);
             }
         } else {
             if (needJUnit)
                 cp.append(params.getJUnitPath());
 
-            if (needTestNG)
+            if (needTestNG) {
                 cp.append(params.getTestNGPath());
+                cp.append(new SearchPath("/usr/share/java/jcommander.jar"));
+            }
         }
 
         // Extras:
@@ -912,8 +916,10 @@
                 File md = workDir.getFile("modules");
                 if (needJUnit)
                     install(params.getJUnitPath(), md);
-                if (needTestNG)
+                if (needTestNG) {
                     install(params.getTestNGPath(), md);
+                    install(new SearchPath("/usr/share/java/jcommander.jar"), md);
+                }
                 mp.append(md);
             }
         } else {
@@ -921,8 +927,10 @@
             if (needJUnit)
                 fp.append(params.getJUnitPath());
 
-            if (needTestNG)
+            if (needTestNG) {
                 fp.append(params.getTestNGPath());
+                fp.append(new SearchPath("/usr/share/java/jcommander.jar"));
+            }
         }
 
         // Extras:
